core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/LongTerms.java

```diff
@@ -91,6 +91,9 @@ public Bucket(long term, long docCount, InternalAggregations aggregations, boole
 
         @Override
         public String getKeyAsString() {
+            if (formatter instanceof ValueFormatter.BooleanFormatter) {
+                return term == 0 ? "false" : "true";
+            }
             return String.valueOf(term);
```

Comment:
I tried that in my first attempt and then I realized that it would break the Bucket.getBucketByKey for the date field. If we are ok to break this then yes the best way to fix the problem would be to use the formatter for all the types in getKeyAsString.